src/main/java/meditracker/library/SearchResult.java

```diff
@@ -0,0 +1,60 @@
+package meditracker.library;
+
+/**
+ * Represents a search result containing details about a medication.
+ */
+public class SearchResult {
+    private final String name;
+    private final String illness;
+    private final String sideEffects;
+
+    /**
+     * Constructs a SearchResult object with the specified medication details.
+     *
+     * @param name        The name of the medication.
+     * @param illness     The illness(es) the medication is used to treat.
+     * @param sideEffects The possible side effects of the medication.
+     */
+    public SearchResult(String name, String illness, String sideEffects) {
+        this.name = name;
+        this.illness = illness;
+        this.sideEffects = sideEffects;
+    }
+
+    /**
+     * Gets the name of the medication.
+     *
+     * @return The name of the medication.
+     */
+    private String getName() {
+        return this.name;
+    }
+
+    /**
+     * Gets the illness(es) the medication is used to treat.
+     *
+     * @return The illness(es) the medication is used to treat.
+     */
+    private String getIllness() {
+        return this.illness;
+    }
+
+    /**
+     * Gets the possible side effects of the medication.
+     *
+     * @return The possible side effects of the medication.
+     */
+    private String getSideEffects() {
+        return this.sideEffects;
+    }
```

Comment:
Private getters might not be useful if used just for `toString` method. However, due the nature of the modifiers for the attributes, might I suggest having these method modifiers to be set to `public` instead for ease of future use.